runners/flink/src/test/java/org/apache/beam/runners/flink/FlinkSavepointTest.java

```diff
@@ -194,7 +196,7 @@ private void runSavepointAndRestore(boolean isPortablePipeline) throws Exception
   private JobID executeLegacy(Pipeline pipeline) throws Exception {
     JobGraph jobGraph = getJobGraph(pipeline);
     flinkCluster.submitJob(jobGraph).get();
-    return jobGraph.getJobID();
+    return waitForJobToBeReady();
```

Comment:
I added this line here, maybe if this PR is accepted and merged we can close #15119